docs/resource/README.md

```diff
@@ -102,7 +102,110 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
+The ID helps to distinguish instances of the same service that exist at the same time
+(e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent
+and stay the same for the lifetime of the service instance, however it is acceptable that
+the ID is ephemeral and changes during important lifetime events for the service
+(e.g. service instance restarts).
+
+If the instance has no inherent unique ID that can be used as the value of this attribute,
+implementations MAY generate a random Version 1 or Version 4
+[RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) UUID. When enough data is available,
+implementations SHOULD use Version 5 and SHOULD use the following UUID as the namespace:
+`4d63009a-8d0f-11ee-aad7-4c796ed8e320`.
+
+UUIDs are typically recommended, as only an opaque yet reproducible value for
+the purposes of identifying a service instance is needed. Similar to what can be seen in the man page for the
+[`/etc/machine-id`](https://www.freedesktop.org/software/systemd/man/machine-id.html) file,
+the underlying data, such as pod name and namespace should be treated as
+confidential by this algorithm, being the user's choice to expose it or not via another resource attribute.
+
+When a UUID v5 is generated, the input SHOULD be prefixed with
+`${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}`, followed by the
+the workload identifier, which should tentatively be stable. The workload identifier is the final piece of
+information making the source of telemetry to be uniquely idenfiable. For instance, it might be the `container.name`
+in containerized or Kubernetes environments. This means that the same service yields the
+same UUID if the same identifier (`host.id`, `/etc/machine-id`, and so on) remains the same. It would still
+yield different results for different services on the same host or namespace. When no namespaces or equivalent fields
+are available, the prefix MUST then be `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}`.
+
+For applications running behind an application server, such as unicorn, a stable identifier is not desirable, as
+each worker thread typically is seen as a different instance.
+
+SDKs SHOULD use the following algorithm when generating `service.instance.id`:
+
+- If the user has provided a `service.instance.id`, via environment
+  variable, configuration or custom resource detection, this MUST take priority over generated IDs. The special value `uuidv4` means
+  that a UUID v4 should be generated.
+- When the language, framework, or platform has a native mechanism allowing workloads to be uniquely idenfiable, this mechanism MAY be used,
+  either directly or as an input when generating UUID v5. A generated UUID v5 is preferable if the provided identifier isn't opaque and can
+  leak unwanted information about the workload (like a node name). An example of such native mechanism is Erlang's `erlang:node()`.
```

Comment:
What's this about leaking unwanted information?  Can you clarify that?

Reply:
It's typically not a good idea to expose a hostname without the explicit request of users. For this reason, machine-id is an opaque UUID, for instance. So, while Erlang's `erlang:node()` provides the uniqueness we want, we should hash it (or use as source of UUID v5) instead of using its raw value.